File: frontend/lib/models/branch.dart
class Branch {
  final String id;
  final String businessId;
  final String name;
  final String? contactPhone;
  final String? address;
  final bool isDefault;
  final String status; // active, inactive, closed
  final String? operatingHoursStart; // HH:MM format
  final String? operatingHoursEnd; // HH:MM format
  final DateTime createdAt;
  final DateTime updatedAt;

  Branch({
    required this.id,
    required this.businessId,
    required this.name,
    this.contactPhone,
    this.address,
    this.isDefault = false,
    this.status = 'active',
    this.operatingHoursStart,
    this.operatingHoursEnd,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Branch.fromJson(Map<String, dynamic> json) {
    return Branch(
      id: json['id'] as String,
      businessId: json['business_id'] as String,
      name: json['name'] as String,
      contactPhone: json['contact_phone'] as String?,
      address: json['address'] as String?,
      isDefault: json['is_default'] as bool? ?? false,
      status: json['status'] as String? ?? 'active',
      operatingHoursStart: json['operating_hours_start'] as String?,
      operatingHoursEnd: json['operating_hours_end'] as String?,
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'business_id': businessId,
      'name': name,
      'contact_phone': contactPhone,
      'address': address,
      'is_default': isDefault,
      'status': status,
      'operating_hours_start': operatingHoursStart,
      'operating_hours_end': operatingHoursEnd,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
    };
  }

  Branch copyWith({
    String? id,
    String? businessId,
    String? name,
    String? contactPhone,
    String? address,
    bool? isDefault,
    String? status,
    String? operatingHoursStart,
    String? operatingHoursEnd,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Branch(
      id: id ?? this.id,
      businessId: businessId ?? this.businessId,
      name: name ?? this.name,
      contactPhone: contactPhone ?? this.contactPhone,
      address: address ?? this.address,
      isDefault: isDefault ?? this.isDefault,
      status: status ?? this.status,
      operatingHoursStart: operatingHoursStart ?? this.operatingHoursStart,
      operatingHoursEnd: operatingHoursEnd ?? this.operatingHoursEnd,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
} 

File: backend/internal/models/branch.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// Branch represents the branch model
type Branch struct {
	ID                  string         `gorm:"primaryKey;type:varchar(36)" json:"id"`
	BusinessID          string         `gorm:"type:varchar(36);not null;index" json:"business_id"`
	Name                string         `gorm:"type:varchar(100);not null" json:"name"`
	ContactPhone        *string        `gorm:"type:varchar(20)" json:"contact_phone,omitempty"`
	Address             *string        `gorm:"type:varchar(255)" json:"address,omitempty"`
	IsDefault           bool           `gorm:"default:false" json:"is_default"`
	Status              string         `gorm:"type:varchar(20);default:'active'" json:"status"`
	OperatingHoursStart *string        `gorm:"type:varchar(5)" json:"operating_hours_start,omitempty"` // HH:MM format
	OperatingHoursEnd   *string        `gorm:"type:varchar(5)" json:"operating_hours_end,omitempty"`   // HH:MM format
	CreatedAt           time.Time      `json:"created_at"`
	UpdatedAt           time.Time      `json:"updated_at"`
	DeletedAt           gorm.DeletedAt `gorm:"index" json:"-"`

	// Relationships
	Business *Business `gorm:"foreignKey:BusinessID" json:"business,omitempty"`
}

// Branch status constants
const (
	BranchStatusActive   = "active"
	BranchStatusInactive = "inactive"
	BranchStatusClosed   = "closed"
)

// TableName specifies the table name for Branch model
func (Branch) TableName() string {
	return "branches"
}

// BeforeCreate hook to generate UUID for branch ID
func (b *Branch) BeforeCreate(tx *gorm.DB) error {
	if b.ID == "" {
		b.ID = generateUUID()
	}
	return nil
}

// CreateBranchRequest represents the create branch request payload
type CreateBranchRequest struct {
	BusinessID          string `json:"business_id" binding:"required"`
	Name                string `json:"name" binding:"required,min=2,max=100"`
	ContactPhone        string `json:"contact_phone,omitempty"`
	Address             string `json:"address,omitempty"`
	IsDefault           bool   `json:"is_default,omitempty"`
	Status              string `json:"status,omitempty"`
	OperatingHoursStart string `json:"operating_hours_start,omitempty"`
	OperatingHoursEnd   string `json:"operating_hours_end,omitempty"`
}

// UpdateBranchRequest represents the update branch request payload
type UpdateBranchRequest struct {
	Name                string `json:"name" binding:"omitempty,min=2,max=100"`
	ContactPhone        string `json:"contact_phone,omitempty"`
	Address             string `json:"address,omitempty"`
	IsDefault           *bool  `json:"is_default,omitempty"`
	Status              string `json:"status,omitempty"`
	OperatingHoursStart string `json:"operating_hours_start,omitempty"`
	OperatingHoursEnd   string `json:"operating_hours_end,omitempty"`
}

// ToModel converts CreateBranchRequest to Branch model
func (r *CreateBranchRequest) ToModel() *Branch {
	branch := &Branch{
		BusinessID: r.BusinessID,
		Name:       r.Name,
		IsDefault:  r.IsDefault,
		Status:     BranchStatusActive,
	}

	if r.ContactPhone != "" {
		branch.ContactPhone = &r.ContactPhone
	}
	if r.Address != "" {
		branch.Address = &r.Address
	}
	if r.Status != "" {
		branch.Status = r.Status
	}
	if r.OperatingHoursStart != "" {
		branch.OperatingHoursStart = &r.OperatingHoursStart
	}
	if r.OperatingHoursEnd != "" {
		branch.OperatingHoursEnd = &r.OperatingHoursEnd
	}

	return branch
}

// IsActive returns true if the branch is active
func (b *Branch) IsActive() bool {
	return b.Status == BranchStatusActive
}
